src/app/components/user-details/user-details.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { UserDetails, UserUpdate } from 'src/app/models/user';
+import { UserService } from 'src/app/services/user.service';
+
+@Component({
+    selector: 'app-user-details',
+    templateUrl: './user-details.component.html',
+    styleUrls: ['./user-details.component.scss']
+})
+export class UserDetailsComponent implements OnInit {
+    loading: boolean = false;
+    editMode: boolean = false;
+
+    userEditForm!: FormGroup;
+    email: FormControl;
+
+    tempDetails = {
+        name: "Username",
+        position: "Position"
+    }
+
+    //until getCurrentlyLoggedUser call is implemented
+    userId: number = 7;
+
+    get controls() {
+        return this.userEditForm.controls
+    }
+
+    constructor(private userService: UserService, private _snackBar: MatSnackBar) { }
+
+    ngOnInit(): void {
+        this.initUserFormControls();
+        this.retrieveUserData();
+    }
+
+    initUserFormControls(): void {
+        this.userEditForm = new FormGroup({
+            name: new FormControl({ value: '', disabled: true }, [Validators.required, Validators.pattern('^[a-zA-Z]+[a-zA-Z ]*')]),
+            position: new FormControl({ value: '', disabled: true }, [Validators.required]),
+            phoneNo: new FormControl({ value: '', disabled: true }, [Validators.pattern('^[0-9]{0,10}')]),
+            email: new FormControl({ value: '', disabled: true }, null)
+        });
+    }
+
+    retrieveUserData(): void {
+        this.loading = true;
+        this.userService.getUserById(this.userId).subscribe(
+            (result) => {
+
+                this.tempDetails.name = result.name;
+                this.tempDetails.position = result.position;
+
+                this.controls.name.setValue(this.ifValueEmpty(result.name));
+                this.controls.position.setValue(this.ifValueEmpty(result.position));
+                this.controls.phoneNo.setValue(this.ifValueEmpty(result.phoneNo));
+                this.controls.email.setValue(this.ifValueEmpty(result.email));
```

Comment:
I think you should find another way to display the fact that a field has no value. Setting a value that means no value makes no sense.